src/mercury_engine_data_structures/file_tree_editor.py

```diff
@@ -83,11 +100,19 @@ def _add_pkg_name_for_asset_id(self, asset_id: AssetId, pkg_name: Optional[str])
         self._files_for_asset_id[asset_id].add(pkg_name)
 
     def _update_headers(self):
+        from mercury_engine_data_structures.version_validation import check_version, get_expected_assetids
```

Comment:
Move this import to the top of the file.

Personally I'd import do just `from mercury_engine_data_structures import version_validation`, kinda got lost looking for a `check_version` function in the current file.